lib/ReactViews/Custom/Chart/ChartPanel.jsx

```diff
@@ -30,15 +29,7 @@ const ChartPanel = createReactClass({
   },
 
   closePanel() {
-    const chartableItems = this.props.terria.catalog.chartableItems;
-    for (let i = chartableItems.length - 1; i >= 0; i--) {
-      const item = chartableItems[i];
-      if (item.isEnabled && defined(item.tableStructure)) {
-        item.tableStructure.columns
-          .filter(column => column.isActive)
-          .forEach(column => column.toggleActive());
-      }
-    }
+    this.props.viewState.chartIsOpen(false);
```

Comment:
Took me ages to figure out why `chartIsOpen` is a function instead of a property. It's defined in `ViewState` using `defineProperty`, right? And usually that would create a property, so you'd write code like `this.props.viewState.chartIsOpen = false`. I _think_ the reason you're not getting a property as expected is because `chartIsOpen` is being passed to `knockout.track` prior to the call to `knockout.defineProperty`. You shouldn't do both, and apparently if you do Knockout does some kinda weird stuff. TL;DR: remove `chartIsOpen` from `knockout.track` and change all the places you're using `chartIsOpen` to treat it like a property instead of a function.

Reply:
Fixed